pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -435,6 +434,16 @@ public void addSegment(ImmutableSegment segment) {
     }
   }
 
+  protected boolean shouldSkipAddSegmentOutOfTTL() {
+    return true;
+  }
+
+  private boolean isOutOfMetadataTTL(IndexSegment segment) {
+    Number maxComparisonValue =
```

Comment:
we can check before reading segment metadata if `_largestSeenComparisonValue.get() != LARGEST_COMPARISON_VALUE_NOT_SET` else return true.